src/ios/OpenTokPlugin.m

```diff
@@ -346,20 +346,111 @@ - (void)subscriberDidConnectToStream:(OTSubscriberKit*)sub{
     NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
     NSString* streamId = sub.stream.streamId;
     [eventData setObject:streamId forKey:@"streamId"];
-    [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData];
-    
+    [self triggerJSEvent: @"subscriberEvents" withType: @"connected" withData: eventData];
+    [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData]; // Backwards compatibility
+}
+- (void)subscriberDidDisconnectFromStream:(OTSubscriberKit*)sub{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    NSString* streamId = sub.stream.streamId;
+    [eventData setObject:streamId forKey:@"streamId"];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"disconnected" withData: eventData];
 }
-- (void)subscriber:(OTSubscriber*)subscrib didFailWithError:(OTError*)error{
+- (void)subscriber:(OTSubscriber*)sub didFailWithError:(OTError*)error{
     NSLog(@"subscriber didFailWithError %@", error);
     NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
-    NSString* streamId = subscrib.stream.streamId;
+    NSString* streamId = sub.stream.streamId;
     NSNumber* errorCode = [NSNumber numberWithInt:1600];
     [eventData setObject: errorCode forKey:@"errorCode"];
     [eventData setObject:streamId forKey:@"streamId"];
     [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData];
 }
+- (void)subscriberVideoDataReceived:(OTSubscriber*)sub{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    //[self triggerJSEvent: @"subscriberEvents" withType: @"videoDataReceived" withData: eventData];
+}
+- (void)subscriberVideoDisabled:(OTSubscriberKit*)sub reason:(OTSubscriberVideoEventReason)reason{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+NSString reasonData = @""
+    if(reason == OTSubscriberVideoEventPublisherPropertyChange) {
+        reasonData = @"publishVideo";
+    } else if(reason == OTSubscriberVideoEventSubscriberPropertyChanged) {
+        reasonData = @"subscribeToVideo";
+    } else if(reason == OTSubscriberVideoEventQualityChanged) {
+        reasonData = @"quality";
+    }
+    
+    [eventData setObject: reasonData forKey:@"reason"];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDisabled" withData: eventData];
+}
+- (void)subscriberVideoDisableWarning:(OTSubscriberKit*)sub{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDisableWarning" withData: eventData];
```

Comment:
why send an empty dictionary? Can we send null since there is no data associated with this event?